### ISSUE TITLE
tech-debt: Qwen3 and LFM2 routers handle input dimensionality inconsistently

### ISSUE THREAD

**Issue by egesabanci** [egesabanci/reap-mlx#92]
## Description

The Qwen3 router **flattens** input hidden states to 2D before passing to the gate, while the LFM2 router passes them **as-is** (potentially 3D `[batch, seq, hidden]`). This inconsistency means architecture-specific assumptions about gate input shape are embedded in the router implementations, not validated.

## Exact Code Citations

**src/reap/router.py, lines 113-118** (Qwen3 flattens):
```python
leading_shape = hidden_states.shape[:-1]
hidden_size = hidden_states.shape[-1]
flat_hidden_states = hidden_states.reshape(-1, hidden_size)
logits = self.gate(flat_hidden_states)
```

**src/reap/router.py, lines 206-211** (LFM2 does NOT flatten):
```python
# (no flattening)
logits = self.gate(hidden_states).astype(mx.float32)
```

## Impact

- If an LFM2 model variant has a gate that expects 2D input, 3D input would produce wrong results (MLX broadcasting might inadvertently "work" but produce baffling logits).
- If a future Qwen3 model variant has a gate with different dimensionality handling, the flattening assumption would break.
- Inconsistency makes it harder to add new router implementations -- developers must guess whether to flatten or not.

## Suggested Fix

Standardize the approach: either always flatten (ensuring 2D gate input) and reshape back, or always pass through and let the gate handle dimensionality. Document the choice explicitly with a validation assertion.
